src/Octoshift/Services/GithubApi.cs

```diff
@@ -491,7 +491,7 @@ public virtual async Task<(string State, string RepositoryName, string FailureRe
         }
     }
 
-    public virtual async Task<string> GetMigrationLogUrl(string org, string repo)
+    public virtual async Task<(string MigrationLogUrl, string MigrationId)?> GetMigrationLogUrl(string org, string repo)
```

Comment:
As i already mentioned we can return a not nullable value tuple here since we are checking the `MigrationLogUrl` anyways.

Reply:
Shouldn't it be nullable for the case so we can signal the case where no migration is found? 

I think there are three cases we need to handle here:

* Migration not found
* Migration found, but no log
* Migrations found with log

It's possible that I'm misunderstanding your comment, so apologies if so!

Reply:
We could change the logic so we return `(null, null)` in the "migration not found" case. That would be an option :)

Reply:
Ah, i think you're suggesting that in the comment below. Does the `default` keyword return `(null, null)`?

Reply:
the default will return (null, null), I am also suggesting to not return a nullable value tuple.

Reply:
Oh my apologies, I totally missed the fact that we decided to return a nullable type to indicate that the migration is missing. So in that case there is no need to change the type but the `default` can def. state it will return the default value of `(string MigrationLogUrl, string MigrationId)?` which is `null` :)